File: tools/patch_deal_ads_authorid.js
// Patch all deal_ads in Firestore to add authorID field (set to listerId)
// Usage: node patch_deal_ads_authorid.js

const admin = require('firebase-admin');
admin.initializeApp({
  credential: admin.credential.cert(require('../caribtap-firebase-adminsdk-fbsvc-f2e46066e1.json'))
});

async function patchDealAds() {
  const db = admin.firestore();
  const adsSnap = await db.collection('deal_ads').get();
  let patched = 0;
  for (const doc of adsSnap.docs) {
    const data = doc.data();
    if (!data.authorID && data.listerId) {
      console.log(`Patching ${doc.id} with authorID: ${data.listerId}`);
      await doc.ref.update({ authorID: data.listerId });
      patched++;
    }
  }
  console.log(`Patch complete. ${patched} documents updated.`);
}

patchDealAds().catch(console.error);

File: tools/patch_deal_ads_authorid.dart
// Patch all deal_ads in Firestore to add authorID field (set to listerId)
// Run this script with `dart run` or in a Dart environment with Firebase Admin SDK

import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';

Future<void> main() async {
  await Firebase.initializeApp();
  final firestore = FirebaseFirestore.instance;
  final ads = await firestore.collection('deal_ads').get();
  for (final doc in ads.docs) {
    final data = doc.data();
    if (!data.containsKey('authorID') && data.containsKey('listerId')) {
      print('Patching ${doc.id} with authorID: ${data['listerId']}');
      await doc.reference.update({'authorID': data['listerId']});
    }
  }
  print('Patch complete.');
}
